app/bridge/server.js

```diff
@@ -90,6 +128,38 @@ function createServer(name, version, methods) {
     addEventListener('message', e => this.onglobalmessage(e.data));
   };
 
+  ServerInternal.prototype.register = function() {
+    debug(this.server.name, ' [connect]');
+    var smuggler = new BroadcastChannel('smuggler');
+    smuggler.postMessage({
+      name: 'register',
+      type: 'server',
+      contract: this.server.name,
+      version: this.server.version,
+    });
+    smuggler.close();
+
+  };
+
+  ServerInternal.prototype.unregister = function() {
+    debug(this.server.name, 'Unregistering server');
+    this.ports.forEach(port => this.disconnectPort(port));
+
+    // empty the array
+    this.ports.length = 0;
+
+    // and now please kill me!
+    var smuggler = new BroadcastChannel('smuggler');
+    smuggler.postMessage({
+      name: 'unregistered',
+      type: 'server',
+      contract: this.server.name,
+      version: this.server.version,
+    });
+    smuggler.close();
+
+  };
```

Comment:
Similar comments that the version for the Client.

Also it seems inconsistent to have a register/unregister functions unexposed to ClientInternal versus explicit properties of ServerInternal here.
Make them simple register/unregister functions too and move them out of the middle of the ServerInternal definition.